src/storage/pooled_storage_manager.h

```diff
@@ -32,6 +32,8 @@ class GPUPooledStorageManager final : public StorageManager {
    */
   GPUPooledStorageManager() {
     reserve_ = dmlc::GetEnv("MXNET_GPU_MEM_POOL_RESERVE", 5);
+    cudaGetDeviceCount(&ndev_);
```

Comment:
The ndev_ could actually be a constant equal to 16, it should not make a big difference. Do you think it would be better?